### DISCUSSION TITLE
How to setup boundary conditions for a loop

### DISCUSSION THREAD

**Discussion by hhy2022** [idaholab/moose#22427] [Q&A General] [answered]
I want to simulate the fluid dynamic for a loop but I am a little confused on the definition on the boundary conditions. I know for a tube problem, we always have a left side and right side boundary conditions. But what if it is for a loop problem? Is there any reference to start with?

Many thanks for any help.

**Comment by GiudGiud** [comment, marked as answer]
Hello

for a loop, in multidimensional problems you ll have boundary conditions for the walls of the loop. 

But there won’t be any inlet or outlet boundary conditions. 

Are you planning to use Navier Stokes for multiD flow or the thermal hydraulics module for 1D flow? We should be able to point you to some examples

Guillaume

**Reply by hhy2022** [reply]
Hi,

I am not using specified module to solve the problem. I want to solve a simplified transport equation for a 1D tube. So I assembled some kernels together. Now I want to move to a loop problem but now I am a litte confused on how to setup the boundary condition?

**Reply by GiudGiud** [reply]
So if you're doing a 1D loop, then I reckon you dont need a boundary condition since there's no boundaries.

You may need a constraint on pressure if you are doing incompressible flow though. You can either pin the value at a point or pin the average of the pressure over the domain.

**Reply by hhy2022** [reply]
That makes sense. Thank you so much.
